pkg/services/pluginsintegration/clientmiddleware/grafana_request_id_header_middleware.go

```diff
@@ -49,17 +50,23 @@ func (m *HostedGrafanaACHeaderMiddleware) applyGrafanaRequestIDHeader(ctx contex
 	}
 
 	// Check if the request is for a datasource that is allowed to have the header
-	target := pCtx.DataSourceInstanceSettings.URL
-
+	dsURL := pCtx.DataSourceInstanceSettings.URL
+	// Only look at the scheme and host, ignore the path
+	dsBaseURL, err := url.Parse(dsURL)
+	if err != nil {
+		m.log.Debug("Failed to parse data source URL", "error", err)
+		return
+	}
+	dsBaseURL.Path = ""
 	foundMatch := false
 	for _, allowedURL := range m.cfg.IPRangeACAllowedURLs {
-		if path.Clean(allowedURL) == path.Clean(target) {
+		if path.Clean(allowedURL) == path.Clean(dsBaseURL.String()) {
```

Comment:
`path.Clean()` has the odd side effect that it removes one `/` from `https://`. I think it wasn't meant to be used with URLs. It's probably OK for our use case, but alternatively we could build an URL from the `allowedURL` and just compare `URL.Scheme` and `URL.Host`. There's still a question of port being present as part of the host, however. We could also split the host into host and port components, I suppose.

WDYT?

Reply:
That makes sense, and I agree comparing on just the parsed components of the URL is a bit more future proof. Reasoning is that, there's also the query string and fragment that can be present on the URL that would conflict if not removed.

If we do decide to compare based on parsed URLs from `m.cfg.IPRangeACAllowedURLs`, it would be good to store the parsed URLs in that value then to avoid parsing for every request.

Reply:
Btw, I think this is an improvement that we can do later, too. That is, we can stick now to the current implementation as long as it covers our immediate needs. I'll leave it up to you :)